tests/flytekit/unit/annotated/test_type_hints.py

```diff
@@ -531,10 +534,18 @@ def t2(a: str) -> str:
     @workflow
     def my_wf(a: int, b: str) -> (int, str):
         x, y = t1(a=a)
-        d = conditional().if_(x == 4).then(t2(a=b)).elif_(x >= 5).then(t2(a=y)).else_().fail("All Branches failed")
+        d = (
+            conditional("test1")
+            .if_(x == 4)
+            .then(t2(a=b))
+            .elif_(x >= 5)
+            .then(t2(a=y))
+            .else_()
+            .fail("All Branches failed")
+        )
         return x, d
 
-    with pytest.raises(AssertionError):
+    with pytest.raises(ValueError):
```

Comment:
does the failure value make it into the ValueError at all?  Just wondering.

Reply:
ya it does, we could make it something else. This is what it is today